bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/console/InboxConsoleCommandExtension.java

```diff
@@ -100,13 +101,49 @@ public void execute(String[] args, Console console) {
                 case SUBCMD_CLEAR:
                     clearInboxEntries(console, inbox.getAll());
                     break;
+                case SUBCMD_REMOVE:
+                    if (args.length > 2) {
+                        if (args[1].equalsIgnoreCase("id")) {
+                            try {
+                                ThingUID thingUID = new ThingUID(args[2]);
+                                List<DiscoveryResult> results = inbox.stream().filter(forThingUID(thingUID))
+                                        .collect(Collectors.toList());
+                                if (results.isEmpty()) {
+                                    console.println("No matching inbox entry could be found.");
+                                } else {
+                                    clearInboxEntries(console, results);
+                                }
+                            } catch (Exception e) {
+                                console.println("'" + args[2] + "' is no valid thing UID.");
+                            }
+                        } else if (args[1].equalsIgnoreCase("type")) {
+                            try {
+                                ThingTypeUID thingTypeUID = new ThingTypeUID(args[2]);
+                                List<DiscoveryResult> results = inbox.stream().filter(forThingTypeUID(thingTypeUID))
+                                        .collect(Collectors.toList());
+                                if (results.isEmpty()) {
+                                    console.println("No matching inbox entry could be found.");
+                                } else {
+                                    clearInboxEntries(console, results);
+                                }
+                            } catch (Exception e) {
+                                console.println("'" + args[2] + "' is no valid thing type.");
+                            }
+                        } else {
+                            console.println(
+                                    "Specify thing id or thing type to remove: inbox remove id <thingUID> OR inbox remove type <thingTypeUID>");
```

Comment:
From a user perspective, the "id" and "type" argument come rather as a surpise - I would usually assume to directly do a `inbox remove xxxxx".
I think that there is no situation where a thingUID is identical to a thingTypeUID, so I would claim that the implementation can be easily changed to drop the "id" and "type word, so that we have a usage like `inbox remove [<thingUID>|<thingTypeUID>]`. Wdyt?

Reply:
Ok but if I don't know if the user wants to fill a tingUID or thingTypeUID, I can't display a precise message.